src/software/jetson_nano/services/network/network.cpp

```diff
@@ -22,9 +24,14 @@ std::tuple<TbotsProto::PrimitiveSet, TbotsProto::World> NetworkService::poll(
     const TbotsProto::RobotStatus& robot_status)
 {
     std::scoped_lock lock{primitive_set_mutex, world_mutex};
-    TbotsProto::RobotStatus new_status = robot_status;
-    new_status.set_last_handled_primitive_set(primitive_set_msg.sequence_number());
-    sender->sendProto(robot_status);
+    // Rate limit sending of proto based on thunderloop freq
+    if (network_ticks / (thunderloop_ticks + 1.0) <=
+        ROBOT_STATUS_BROADCAST_RATE_HZ / (CONTROL_LOOP_HZ + 1.0))
```

Comment:
This is a bit confusing to read and understand. Maybe refactoring `ROBOT_STATUS_BROADCAST_RATE_HZ / (CONTROL_LOOP_HZ + 1.0)` to a constant with a descriptive name (something related to ratio?!) could help